src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixLegacyResourceDesignerStep.cs

```diff
@@ -0,0 +1,158 @@
+		{
+			this.cache = cache;
+		}
+
+		readonly
+#endif
+		IMetadataResolver cache;
+		AssemblyDefinition designerAssembly = null;
+		TypeDefinition designerType = null;
+		Dictionary<string, MethodDefinition> lookup;
+
+		protected override void EndProcess ()
+		{
+			if (designerAssembly != null) {
+				LogMessage ($"  Setting Action on {designerAssembly.Name} to Link.");
+				Annotations.SetAction (designerAssembly, AssemblyAction.Link);
+			}
+		}
+
+		protected override void LoadDesigner ()
+		{
+			if (designerAssembly != null)
+				return;
+			var designerNameAssembly = AssemblyNameReference.Parse ($"{DesignerAssemblyName}, Version=1.0.0.0");
+			try {
+				designerAssembly = Resolve (designerNameAssembly);
+			} catch (Mono.Cecil.AssemblyResolutionException) {
+			} catch (System.IO.FileNotFoundException) {
+			}
+			if (designerAssembly == null) {
+				LogMessage ($"   Did not find {DesignerAssemblyName}");
+				return;
+			}
+			designerType = designerAssembly.MainModule.GetTypes ().FirstOrDefault (x => x.FullName == $"{DesignerAssemblyName}.Resource");
+			if (designerType == null) {
+				LogMessage ($"   Did not find {DesignerAssemblyName}.Resource type. It was probably linked out.");
+				return;
+			}
+			lookup = BuildResourceDesignerPropertyLookup (designerType);
+			return;
+		}
+		internal override bool ProcessAssemblyDesigner (AssemblyDefinition assembly)
+		{
+			if (designerAssembly == null || designerType == null) {
+				LogMessage ($"   Not using {DesignerAssemblyName}");
+				return false;
+			}
+
+			if (!FindResourceDesigner (assembly, mainApplication: false, out TypeDefinition designer, out CustomAttribute designerAttribute)) {
+				LogMessage ($"   {assembly.Name.Name} has no designer. ");
+				return false;
+			}
+
+			LogMessage ($"   {assembly.Name.Name} has a designer. ");
+			LogMessage ($"   BaseType: {designer.BaseType.FullName}. ");
+			if (designer.BaseType.FullName == $"{DesignerAssemblyName}.Resource") {
+				LogMessage ($"   {assembly.Name.Name} has already been processed. ");
+				return false;
+			}
+
+			LogMessage ($"    Adding reference {designerAssembly.Name.Name}.");
+			assembly.MainModule.AssemblyReferences.Add (designerAssembly.Name);
+			var importedDesignerType = assembly.MainModule.ImportReference (designerType.Resolve ());
+
+			LogMessage ($"    FixupAssemblyTypes {assembly.Name.Name}.");
+			// now replace all ldsfld with a call to the property get_ method.
+			FixupAssemblyTypes (assembly, designer);
+
+			LogMessage ($"    ClearDesignerClass {assembly.Name.Name}.");
+			// then clean out the designer.
+			ClearDesignerClass (designer, completely: true);
+			designer.BaseType = importedDesignerType;
+			return true;
+		}
+
+		Dictionary<string, MethodDefinition> BuildResourceDesignerPropertyLookup (TypeDefinition type)
+		{
+			LogMessage ($"     Building Designer Lookups for {type.FullName}");
+			var output = new Dictionary<string, MethodDefinition> (StringComparer.OrdinalIgnoreCase);
```

Comment:
Should this use `OrdinalIgnoreCase`? Could you have properties `Foo` and `fOo`?

In some cases, does Android preserve casing like styles/themes?

Reply:
The comparisons where are restricted to those in the Resource Designer subclasses. 
When we use the lookup later in code we are also restricting the IL so that is MUST include the `Resource Designer namespace`. 

For example we look for lines like

```
IL_0068: ldsfld int32 Xamarin.Forms.Platform.Android.Resource/Layout::Toolbar
```
We will specifically lookf or `Xamarin.Forms.Platform.Android.Resource/`before we then check the property via the lookup.

While we can do this lookup via an `Ordinal`, I think `OrdinalIgnoreCase` is quicker, which is why I used it. I think the other checks are enough to make sure we are fixing the right code. But I could be wrong about that....

@jonp what do you think?

Reply:
`.Ordinal` *should* be faster than `.OrdinalIgnoreCase`.  I'm not sure why you would think that, or how `.OrdinalIgnoreCase` could ever be faster…

Generally, if a name comes from the filesystem, we need `.OrdinalIgnoreCase`.  If it comes from IL or C#, it can be `.Ordinal`.